components/core/tests/test-ffi_Value.cpp

```diff
@@ -0,0 +1,172 @@
+requires(std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>
+         || std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+[[nodiscard]] auto get_encoded_text_ast(std::string_view text
+) -> clp::ir::EncodedTextAst<encoded_variable_t>;
+
+template <typename encoded_variable_t>
+requires(std::is_same_v<encoded_variable_t, eight_byte_encoded_variable_t>
+         || std::is_same_v<encoded_variable_t, four_byte_encoded_variable_t>)
+auto get_encoded_text_ast(std::string_view text) -> clp::ir::EncodedTextAst<encoded_variable_t> {
+    string logtype;
+    vector<encoded_variable_t> encoded_vars;
+    vector<int32_t> dict_var_bounds;
+    REQUIRE(clp::ffi::encode_message(text, logtype, encoded_vars, dict_var_bounds));
+    REQUIRE(((dict_var_bounds.size() % 2) == 0));
+
+    vector<string> dict_vars;
+    for (size_t i{0}; i < dict_var_bounds.size(); i += 2) {
+        auto const begin_pos{static_cast<size_t>(dict_var_bounds[i])};
+        auto const end_pos{static_cast<size_t>(dict_var_bounds[i + 1])};
+        dict_vars.emplace_back(text.cbegin() + begin_pos, text.cbegin() + end_pos);
+    }
+
+    return clp::ir::EncodedTextAst<encoded_variable_t>{logtype, dict_vars, encoded_vars};
+}
+}  // namespace
+
+TEST_CASE("ffi_Value_basic", "[ffi][Value]") {
+    Value const null_value;
+    REQUIRE(null_value.is_null());
+    REQUIRE_FALSE(null_value.is<value_int_t>());
+    REQUIRE_FALSE(null_value.is<value_float_t>());
+    REQUIRE_FALSE(null_value.is<value_bool_t>());
+    REQUIRE_FALSE(null_value.is<string>());
+    REQUIRE_FALSE(null_value.is<EightByteEncodedTextAst>());
+    REQUIRE_FALSE(null_value.is<FourByteEncodedTextAst>());
+    REQUIRE_THROWS(null_value.get_immutable_view<value_int_t>());
+    REQUIRE_THROWS(null_value.get_immutable_view<value_float_t>());
+    REQUIRE_THROWS(null_value.get_immutable_view<value_bool_t>());
+    REQUIRE_THROWS(null_value.get_immutable_view<string>());
+    REQUIRE_THROWS(null_value.get_immutable_view<EightByteEncodedTextAst>());
+    REQUIRE_THROWS(null_value.get_immutable_view<FourByteEncodedTextAst>());
+
+    constexpr value_int_t cIntVal{1000};
+    Value const int_value{cIntVal};
+    REQUIRE(int_value.is<value_int_t>());
+    REQUIRE((int_value.get_immutable_view<value_int_t>() == cIntVal));
+    REQUIRE_FALSE(int_value.is_null());
+    REQUIRE_FALSE(int_value.is<value_float_t>());
+    REQUIRE_FALSE(int_value.is<value_bool_t>());
+    REQUIRE_FALSE(int_value.is<string>());
+    REQUIRE_FALSE(int_value.is<EightByteEncodedTextAst>());
+    REQUIRE_FALSE(int_value.is<FourByteEncodedTextAst>());
+    REQUIRE_THROWS(int_value.get_immutable_view<value_float_t>());
+    REQUIRE_THROWS(int_value.get_immutable_view<value_bool_t>());
+    REQUIRE_THROWS(int_value.get_immutable_view<string>());
+    REQUIRE_THROWS(int_value.get_immutable_view<EightByteEncodedTextAst>());
+    REQUIRE_THROWS(int_value.get_immutable_view<FourByteEncodedTextAst>());
```

Comment:
We can extract this into a method that does something like:

```cpp
if constexpr (std::is_same_v<value_int_t, Type>) {
    REQUIRE(value.get_immutable_view<Type>() == typed_value);
} else {
    REQUIRE_THROWS(value.get_immutable_view<value_int_t>());
}
```

for each type, right?